### PR TITLE
Add Test Cases

### PR DESCRIPTION
**Pull request by Charles1026**
Test cases added for consult command, consult command parser and command history.

More test cases added for other classes too

### CODE REVIEW

File: src/test/java/seedu/address/ui/CommandHistoryTest.java [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+    private CommandHistory commandHistory = new CommandHistory();
+    private final String[] commands = {"view 1", "delappt 2", "consult 1", "edit 4 n/Charles", "view 1"};
+
+    @Test
+    void addCommand_addFirstCommand_success() {
+        commandHistory.addCommand(commands[0]);
+        commandHistory.previousCommand();
+        assertEquals(commandHistory.getCommand().orElse(""), commands[0]);
+
+        commandHistory = new CommandHistory();
```
Comment (PeiYee88) [inline src/test/java/seedu/address/ui/CommandHistoryTest.java:19]: You can either move this line out or remove this because it does not help with the current test case.

File: src/test/java/seedu/address/ui/CommandHistoryTest.java [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+    }
+
+    @Test
+    void addCommand_addCommandPointsToLatestCommand_success() {
+        commandHistory.addCommand(commands[0]);
```
Comment (PeiYee88) [inline src/test/java/seedu/address/ui/CommandHistoryTest.java:24]: in addition to previous comment, can do `commandHisory = new CommandHistory();` at the start of the test case

**Review by PeiYee88 — COMMENTED** [review]
Added comments for commandHistoryTest

File: src/test/java/seedu/address/ui/CommandHistoryTest.java [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+        commandHistory.previousCommand();
+        commandHistory.previousCommand();
+        assertEquals(commandHistory.getCommand().orElse(""), commands[0]);
+
+        commandHistory = new CommandHistory();
+
+    }
+
+    @Test
+    void nextCommand_capsAtArraySize_success() {
```
Comment (PeiYee88) [inline src/test/java/seedu/address/ui/CommandHistoryTest.java:62]: Perhaps can reword it to`capsAtCommandHistoryArraySize`. Now sounds confusing because the commands and commandHistory are both in arrays.

File: src/test/java/seedu/address/model/person/EmailTest.java [hunk 4]
```diff
@@ -65,4 +65,14 @@ public void isValidEmail() {
...
         assertTrue(Email.isValidEmail("e1234567@u.nus.edu")); // more than one period in domain
     }
+
+    @Test
+    void testHashCode() {
+        Email firstEmail = new Email("PeterJack_1190@example.com");
+        Email firstEmailCopy = new Email("PeterJack_1190@example.com");
+        Email secondEmail = new Email("peterjack@example.com");
+
+        assertTrue(firstEmail.hashCode() == firstEmailCopy.hashCode());
```
Comment (PeiYee88) [inline src/test/java/seedu/address/model/person/EmailTest.java:75]: can add a test case to check if the hashCode is the same across the same object like `firstEmail.hashcode() == firstEmail.hashcode()` to ensure the hashCode function is not generated randomly.

**Review by PeiYee88 — COMMENTED** [review]
Reviewed model test.

**Review by PeiYee88 — APPROVED** [review]
Thanks for adding test cases!

**Review by Ryan-Tan00 — APPROVED** [review]
LGTM
